core/src/Cabana_Experimental_NeighborList.hpp

```diff
@@ -0,0 +1,207 @@
+    typename stdcxx20::remove_cvref_t<Slice>::size_type last,
+    typename stdcxx20::remove_cvref_t<Slice>::value_type radius )
+{
+    return Impl::SubsliceAndRadius<stdcxx20::remove_cvref_t<Slice>>{
+        std::forward<Slice>( slice ), first, last, radius};
+}
+} // namespace Impl
+} // namespace Experimental
+} // namespace Cabana
+
+namespace ArborX
+{
+namespace Traits
+{
+template <typename Slice>
+struct Access<Slice, PrimitivesTag,
+              std::enable_if_t<Cabana::is_slice<Slice>::value>>
+{
+    using memory_space = typename Slice::memory_space;
+    using size_type = typename Slice::size_type;
+    static size_type size( Slice const &x ) { return x.size(); }
+    static KOKKOS_FUNCTION Point get( Slice const &x, size_type i )
+    {
+        return {x( i, 0 ), x( i, 1 ), x( i, 2 )};
+    }
+};
+template <typename SliceLike>
+struct Access<SliceLike, PredicatesTag>
+{
+    using memory_space = typename SliceLike::memory_space;
+    using size_type = typename SliceLike::size_type;
+    static size_type size( SliceLike const &x ) { return x.last - x.first; }
+    static KOKKOS_FUNCTION auto get( SliceLike const &x, size_type i )
+    {
+        assert( i < size( x ) );
+        auto const point =
+            Access<typename SliceLike::slice_type, PrimitivesTag>::get(
+                x.slice, x.first + i );
+        return attach( intersects( Sphere{point, x.radius} ), (int)i );
+    }
+};
+} // namespace Traits
+} // namespace ArborX
+
+namespace Cabana
+{
+namespace Experimental
+{
+namespace Impl
+{
+
+// Custom callback for ArborX::BVH::query()
+template <typename Tag>
+struct NeighborDiscriminator;
+
+template <>
+struct NeighborDiscriminator<Cabana::FullNeighborTag>
+{
+    using tag = ArborX::Details::InlineCallbackTag;
+    template <typename Predicate, typename OutputFunctor>
+    KOKKOS_FUNCTION void operator()( Predicate const &predicate, int i,
+                                     OutputFunctor const &out ) const
+    {
+        if ( getData( predicate ) != i ) // discard self-collision
+        {
+            out( i );
+        }
+    }
+};
+
+template <>
+struct NeighborDiscriminator<Cabana::HalfNeighborTag>
+{
+    using tag = ArborX::Details::InlineCallbackTag;
+    template <typename Predicate, typename OutputFunctor>
+    KOKKOS_FUNCTION void operator()( Predicate const &predicate, int i,
+                                     OutputFunctor const &out ) const
+    {
+        if ( getData( predicate ) > i )
+        {
+            out( i );
+        }
+    }
+};
+
+template <typename MemorySpace, typename Tag>
+struct CrsGraph
+{
+    Kokkos::View<int *, MemorySpace> col_ind;
+    Kokkos::View<int *, MemorySpace> row_ptr;
+    typename MemorySpace::size_type shift;
+    typename MemorySpace::size_type total;
+};
+
+} // namespace Impl
+
+template <typename DeviceType, typename Slice, typename Tag>
+auto makeNeighborList( Tag, Slice const &coordinate_slice,
```

Comment:
I think we need something to differentiate this from other neighbor construction

Reply:
What name do you suggest?

Reply:
`makeArborXList` or `makeTreeList`?

Reply:
It constructs a neighbor list so the name is appropriate in my opinion.

We could
* either append something to denote what algo/library (but don't see the point because it would not be helpful in a generic context),
* or pass yet another tag so it can be overloaded for the Verlet list.

Reply:
After discussion, I agree with merging this as is and improving/combining the neighbor list storage/generation in a later PR